geomstats/geometry/discrete_surfaces.py

```diff
@@ -320,8 +343,567 @@ def surface_metric_matrices(self, point):
+            evaluated at a vector field defined on the surface point.
+
+            Parameters
+            ----------
+            tangent_vec : array-like, shape=[..., n_vertices, 3]
+                Tangent vector to the manifold at the base point that is the
+                triangulated surface. This tangent vector is a vector field
+                on the triangulated surface.
+
+            Returns
+            -------
+            laplacian_at_tangent_vec: array-like, shape=[..., n_vertices, 3]
+                Mesh Laplacian operator of the triangulated surface applied
+                to one its tangent vector tangent_vec.
+            """
+            to_squeeze = False
+            if tangent_vec.ndim == 2:
+                tangent_vec = gs.expand_dims(tangent_vec, axis=0)
+                to_squeeze = True
+            n_tangent_vecs = len(tangent_vec)
+            tangent_vec_diff = (
+                tangent_vec[:, id_vertices[0]] - tangent_vec[:, id_vertices[1]]
+            )
+            values = gs.einsum(
+                "bd,nbd->nbd", gs.stack([gs.flatten(cot)] * 3, axis=1), tangent_vec_diff
+            )
+
+            laplacian_at_tangent_vec = gs.zeros((n_tangent_vecs, n_vertices, 3))
+            id_vertices_201_repeated = gs.tile(id_vertices[1, :], (n_tangent_vecs, 1))
+
+            for i_dim in range(3):
+                laplacian_at_tangent_vec[:, :, i_dim] = gs.scatter_add(
+                    input=laplacian_at_tangent_vec[:, :, i_dim],
+                    dim=1,
+                    index=id_vertices_201_repeated,
+                    src=values[:, :, i_dim],
+                )
+            return (
+                gs.squeeze(laplacian_at_tangent_vec, axis=0)
+                if to_squeeze
+                else laplacian_at_tangent_vec
+            )
+
+        return _laplacian
+
+
+class ElasticMetric(RiemannianMetric):
+    """Elastic metric defined by a family of second order Sobolev metrics.
+
+    Each individual discrete surface is represented by a 2D-array of shape `[
+    n_vertices, 3]`. See [HSKCB2022]_ for details.
+
+    The parameters a0, a1, b1, c1, d1, a2 (detailed below) are non-negative weighting
+    coefficients for the different terms in the metric.
+
+    Parameters
+    ----------
+    space : DiscreteSurfaces
+        Instantiated DiscreteSurfaces manifold.
+    a0 : float
+        First order parameter.
+    a1 : float
+        Stretching parameter.
+    b1 : float
+        Shearing parameter.
+    c1 : float
+        Bending parameter.
+    d1 : float
+        Additonal first order parameter.
+    a2 : float
+        Second order parameter.
+
+    References
+    ----------
+    .. [HSKCB2022] "Elastic shape analysis of surfaces with second-order
+        Sobolev metrics: a comprehensive numerical framework".
+        arXiv:2204.04238 [cs.CV], 25 Sep 2022
+    """
+
+    def __init__(self, space, a0, a1, b1, c1, d1, a2):
```

Comment:
is there any meaningful defaults for this? it is a bit cumbersome to always have to set up 6 parameters.